File: src/de/sciencecomputing/venuside/ContextFileType.java
package de.sciencecomputing.venuside;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import javax.swing.*;

public class ContextFileType extends LanguageFileType {
    public static final ContextFileType INSTANCE = new ContextFileType();

    private ContextFileType() {
        super(ContextLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Context file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "scVENUS context file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return Icons.CONTEXT;
    }
}

File: src/de/sciencecomputing/venuside/ContextFileType.kt
package de.sciencecomputing.venuside

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

object ContextFileType : LanguageFileType(ContextLanguage) {
    override fun getName(): String = "Context file"
    override fun getDescription(): String = "scVENUS context file"
    override fun getDefaultExtension(): String = ""
    override fun getIcon(): Icon? = Icons.CONTEXT
}
